src/python/pants/backend/codegen/protobuf/lint/buf/format_rules.py

```diff
@@ -91,35 +72,29 @@ async def setup_buf_format(request: BufFormatRequest, buf: BufSubsystem) -> Setu
         "-w",
         *buf.format_args,
         "--path",
-        ",".join(source_files_snapshot.files),
+        ",".join(request.snapshot.files),
     ]
     process = Process(
         argv=argv,
         input_digest=input_digest,
-        output_files=source_files_snapshot.files,
+        output_files=request.snapshot.files,
         description=f"Run buf format on {pluralize(len(request.field_sets), 'file')}.",
         level=LogLevel.DEBUG,
         env={
             "PATH": binary_shims.bin_directory,
         },
     )
-    return Setup(process, original_snapshot=source_files_snapshot)
+    return process
 
 
 @rule(desc="Format with buf format", level=LogLevel.DEBUG)
 async def run_buf_format(request: BufFormatRequest, buf: BufSubsystem) -> FmtResult:
     if buf.skip_format:
         return FmtResult.skip(formatter_name=request.name)
-    setup = await Get(Setup, BufFormatRequest, request)
-    result = await Get(ProcessResult, Process, setup.process)
+    process = await Get(Process, BufFormatRequest, request)
+    result = await Get(ProcessResult, Process, process)
```

Comment:
```suggestion
    result = await Get(ProcessResult, BufFormatRequest, request)
```
(here and elsewhere probably)

Reply:
Oh hmm very cool!